src/jwtserviceauth.ts

```diff
@@ -0,0 +1,203 @@
+      body?: unknown
+    ) => Promise<AxiosResponse>,
+    options: Options = {}
+  ) {
+    this.requestHandler = httpRequestHandler || defaultHttpRequestHandler
+    this.authEndpoint = options.endpoint || null
+    this.command = options.command || 'gcloud'
+  }
+
+  public async getGithubAccessToken(
+    privateKey: string,
+    appId: number,
+    installationId: number,
+    appName?: string,
+    options: Options = {}
+  ): Promise<AccessToken> {
+    const expires = options.expires ? options.expires : 600
+
+    // Create JWT auth token
+    const unixNow = Math.floor(new Date().getTime() / 1000)
+    const jwtHeader = {
+      typ: 'JWT',
+      alg: 'RS256'
+    }
+    const jwtBody = {
+      iat: unixNow,
+      exp: unixNow + expires,
+      iss: appId
+    }
+
+    const jwt = JwtUtils.encode(privateKey, jwtHeader, jwtBody)
+
+    // Fetch access token for installation
+    const res = await this.requestHandler(
+      'POST',
+      this.authEndpoint || `https://api.github.com/app/installations/${installationId}/access_tokens`,
+      {
+        Authorization: 'Bearer ' + jwt,
+        'User-Agent': appName ? appName : 'jwtutils',
+        Accept: 'application/vnd.github.machine-man-preview+json'
+      },
+      undefined
+    )
+    if (res && res.status === 201) {
+      const authResponse = res.data
+      const now = new Date().getTime()
+      const expiresAt = new Date(authResponse.expires_at).getTime()
+      return {
+        accessToken: authResponse.token,
+        expiresIn: Math.ceil((expiresAt - now) / 1000),
+        expiresAt: expiresAt
+      }
+    } else {
+      throw new JwtServiceAuthError(`Fetching github access token returned no response`)
+    }
+  }
+
+  public async getGoogleAccessTokenFromGCloudHelper(): Promise<AccessToken> {
+    const resultPromise = ProcessUtils.runProcessAsync(this.command, ['config', 'config-helper', '--format=json'], {
+      closeStdin: true
+    })
+
+    return await resultPromise.then(result => {
+      const config = JSON.parse(result.stdout)
+      const now = new Date().getTime()
+      const expiresAt = new Date(config.credential.token_expiry).getTime()
+      return {
+        accessToken: config.credential.access_token,
+        expiresIn: Math.ceil((expiresAt - now) / 1000),
+        expiresAt: expiresAt
+      }
+    })
+  }
+
+  public async getGoogleAccessToken(
+    keyFileData: string,
+    scopes: string[] | number | null = null,
+    options: Options = {}
+  ): Promise<AccessToken> {
+    const mergedConfig = Object.assign({}, options, {
+      endpoint: this.authEndpoint || 'https://www.googleapis.com/oauth2/v4/token'
+    })
+    return await this._getGoogleAccessToken(this.requestHandler, keyFileData, scopes, mergedConfig)
+  }
+
+  private async _getGoogleAccessToken(
+    httpRequestHandler: (
+      method: string,
+      url: string,
+      headers?: Record<string, string | number>,
+      body?: unknown
+    ) => Promise<AxiosResponse | null>,
```

Comment:
It seems like this function signature is repeated 7 other places, so perhaps it would make sense to make a type out of it